### PR TITLE
[CImGui] cimplot support (imgui v1.76)

### PR DESCRIPTION
**Pull request by wsphillips**
Proposal to include cimplot sources in the build. If approved, I'll move on to include the headers in the generators for the `LibCImGui` module of `CImGui.jl` This ~~looks as though it builds and links appropriately locally~~ builds via cmake locally, but I'm a somewhat C/cmake novice so a draft pull in case I missed something.

### CODE REVIEW

**Comment by wsphillips** [comment]
Any input on what's missing to make the Windows and Apple builds happy appreciated :disappointed:

File: C/CImGui/bundled/wrapper/CMakeLists.txt [hunk 1]
```diff
@@ -43,4 +43,23 @@ install(TARGETS cimgui_helper
         INCLUDES DESTINATION include
         PUBLIC_HEADER DESTINATION include)
 
+add_library(cimplot SHARED)
+
+target_sources(cimplot PRIVATE ./cimplot/cimplot.cpp)
+target_sources(cimplot PRIVATE ./cimplot/implot/implot.cpp)
+target_sources(cimplot PRIVATE ./cimplot/implot/implot_demo.cpp)
+
```
Comment (Gnimuc) [inline C/CImGui/bundled/wrapper/CMakeLists.txt:51]:
```suggestion
target_sources(cimplot PRIVATE ../imgui/imgui.cpp)
```

File: C/CImGui/bundled/wrapper/CMakeLists.txt [hunk 2]
```diff
@@ -43,4 +43,23 @@ install(TARGETS cimgui_helper
...
+target_include_directories(cimplot PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/cimplot
+    ${CMAKE_CURRENT_SOURCE_DIR}/cimplot/implot)
+
+target_link_libraries(cimplot cimgui)
```
Comment (Gnimuc) [inline C/CImGui/bundled/wrapper/CMakeLists.txt:55]: Since we added the imgui source code above, `cimplot` now can work without linking to cimgui.

**Comment by Gnimuc** [comment]
The default "exporting strategy" of mingw is lazy. When compiling cimgui, those imgui internal symbols are not exported. Since imgui functions are directly used in cimplot, when linking cimplot to cimgui, the linker complains about the missing symbols. One way to fix this is to use `-Wl,--export-all-symbols -Wl,-no-undefined -Wl,--enable-runtime-pseudo-reloc` linker flags, another way is to directly include imgui source code in cimplot and not link to cimgui.

As cimgui and cimplot are two separately maintained projects, there will be two version numbers, so it would be better to add cimplot as a new jll project. As long as they based on the same imgui version, it would be fine.

**Comment by wsphillips** [comment]
I will try to get this going as a companion/separate jll repo

**Pull request closed without merging**